docs/proposals/ldap-group-sync.md

```diff
@@ -0,0 +1,295 @@
+  member: (5)
+    key: "DN=%s,OU=Users"
+memberAttribute: "member" (6)
+groupFields: (7)
+    - name: "CN"
+    - uid: "DN"
+    - annotations:
+      - syncrecord: "syncRecord"
+```
+1. The filter that will retrieve all of the first-class groups that this sync will find relevant.
+* The the specific filter that will, when conjoined with the common filter, return only a specific group
+* The method for extracting data from an OpenShift `Group` to populate the specific filter
+* The directions for building the value of the specific filter from an LDAP group name
+* The the specific filter that will return only a specific member when the member annotation from a first-class group is injected and the resulting filter is conjoined with the common filter
+* The attribute on the first-class group entries in the LDAP server that holds their members
+* The specification for how to transfer attributes from a first-class group entity on the LDAP server to fields in an OpenShift `Group`
+
+##  Determining Which Groups To Sync
+An `LDAPGroupLister` determines what LDAP groups needs to be synced and outputs the result as a set of strings which encode the value of the unique filter needed to determine those groups from the LDAP server. For sync jobs where an LDAP request is necessary to determine which groups to sync, the `LDAPGroupLister` will populate the cache used by the `LDAPGroupDataExtractor` as the `LDAPDataSource`.
+
+```go
+// LDAPGroupLister lists the specific group filters for all groups that should be synced
+type LDAPGroupLister interface {
+  ListGroups() (groupFilters util.StringSet, err error)
+}
+```
+
+AN `LDAPGroupMapper` determines what OpenShift `Group` is being referred to by a filter designed for an LDAP group. This mapper finds the OpenShift resource to be updated or creates one if none exists yet.
+```go
+// LDAPGroupMapper maps a set of metainformation to an OpenShift Group
+type LDAPGroupMapper interface{
+  MapLDAPGroup(meta *kapi.ObjectMeta) (group *Group)
+}
+```
+
+### Determining What LDAP Groups Exist
+If the LDAP schema has first-class groups, then determining which groups exist is simple once the LDAP request for all groups is run. If the LDAP schema encodes groups as attributes for first-class users, then determining the list of group names needs to be done from the result of the LDAP query for all users. This query is cached for the `GroupDataExtractor` in this case.
+
+### Forming LDAP Queries for OpenShift `Groups`
+All LDAP requests will contain a conjunction of two filters: a filter that matches all of the LDAP groups or LDAP users that are relevant, and a second filter that, when conjoined with the first matches only the specific LDAP group or set of LDAP members that define a group's membership. An `LDAPFilterGenerator` is used to map from an OpenShift `Group` the filter that will be used to recover it's mirror from the LDAP server.
+
+```go
+// LDAPFilterGenerator generates a filter to match a Group, given a Group
```

Comment:
See, this is why you qualify.  generates a filter to match an LDAP group given an openshift group?  If the document doubles in length, then double it.